ui/page/main_page.go

```diff
@@ -218,20 +231,51 @@ func (mp *MainPage) OnNavigatedTo() {
 			go mp.WL.MultiWallet.Politeia.Sync()
 		}
 	}
+
+	mp.updateExchangeSetting()
+
+	mp.UpdateBalance()
 }
 
 func (mp *MainPage) setLanguageSetting() {
 	langPre := mp.WL.Wallet.ReadStringConfigValueForKey(load.LanguagePreferenceKey)
 	values.SetUserLanguage(langPre)
 }
 
-func (mp *MainPage) UpdateBalance(fetchExch bool) {
-	if len(mp.totalBalanceUSD) == 0 || fetchExch {
-		load.GetUSDExchangeValue(&mp.dcrUsdtBittrex)
-	}
+func (mp *MainPage) updateExchangeSetting() {
 	currencyExchangeValue := mp.WL.Wallet.ReadStringConfigValueForKey(dcrlibwallet.CurrencyConversionConfigKey)
 	mp.usdExchangeSet = currencyExchangeValue == values.USDExchangeValue
+	if mp.usdExchangeSet {
+		go mp.fetchExchangeValue()
+	}
+}
+
+func (mp *MainPage) fetchExchangeValue() {
+	attempts, err := components.RetryFunc(components.RetryAttempts, 2*time.Second, func() error {
```

Comment:
It doesn't make much sense to me for `components` to have a `RetryAttempts` property - how many times something should be retried is best defined by the caller of the function. I'd rather declare and use a `const fetchExchangeRateRetries` field in this file.